import/index_java/src/org/vufind/index/MultiFormatCalculator.java

```diff
@@ -0,0 +1,448 @@
+package org.vufind.index;
+/**
+ * Alternate, multi-valued format determination logic.
+ *
+ * Copyright (C) Villanova University 2018.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+import org.marc4j.marc.Record;
+import org.marc4j.marc.ControlField;
+import org.marc4j.marc.DataField;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Alternate, multi-valued format determination logic.
+ */
+public class MultiFormatCalculator
+{
+    /**
+     * Determine whether a record cannot be a book due to findings in 007.
+     *
+     * @param char formatCode
+     * @return boolean
+     */
+    protected boolean definitelyNotBookBasedOn007(char formatCode) {
+        switch (formatCode) {
+            // Things that are not books: filmstrips/transparencies (G),
+            // pictures (K) and videos/films (M, V):
+            case 'G':
+            case 'K':
+            case 'M':
+            case 'V':
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Determine whether a record cannot be a book due to findings in leader
+     * and fixed fields.
+     *
+     * @param char formatCode
+     * @param ControlField fixedField
+     * @return boolean
+     */
+    protected boolean definitelyNotBookBasedOnRecordType(char recordType, ControlField fixedField) {
+        switch (recordType) {
+            case 'M':
+                // If this is a computer file containing numeric data, it is not a book:
+                if (getTypeOfComputerFile(fixedField) == 'A') {
+                    return true;
+                }
+                break;
+            case 'J':
+            case 'R':
+                // Music recordings (J) and Physical objects (R) are not books.
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Return the best format string based on codes extracted from 007; return
+     * blank string for ambiguous/irrelevant results.
+     *
+     * @param char formatCode
+     * @param char formatCode2
+     * @param char formatCode5
+     * @return String
+     */
+    protected String getFormatFrom007(char formatCode, char formatCode2, char formatCode5) {
```

Comment:
Could this method get the full 007 passed in? I think that'd be more clear and future-proof.

Reply:
Done!